javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/PatternVariableVisitor.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2015-2016 Federico Tomassetti
+ * Copyright (C) 2017-2024 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+package com.github.javaparser.symbolsolver.javaparsermodel;
+
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.expr.*;
+import com.github.javaparser.ast.visitor.GenericVisitorWithDefaults;
+import java.util.*;
+
+public class PatternVariableVisitor extends GenericVisitorWithDefaults<PatternVariableResult, Void> {
+
+    private static PatternVariableVisitor patternVariableVisitor = null;
+
+    private PatternVariableVisitor() {}
+
+    public static PatternVariableVisitor getInstance() {
```

Comment:
Why create a singleton?

Reply:
My reasoning was that there's no reason not to do it since it doesn't hold any state that would cause issues and by having it be a singleton, we skip the cost of creating a new instance with every symbol resolution. It's largely a stylistic choice though since I don't think it would have a noticeable performance impact to create new instances when needed. I'd be happy to refactor this if it's inconsistent with the rest of the project

Reply:
There is no state in this visitor, but the cost of creating the visitor occasionally has no impact on performance. Nor is this a widespread pattern in JP. You can replace this pattern by creating an instance of the visitor.

Reply:
Done!